env.go

```diff
@@ -80,6 +80,8 @@ type Environment interface {
 	AddCloser(func())
 	// Close shutdowns isolated environment and cleans its resources.
 	Close()
+	// WSL2 returns true if the environment is WSL2.
+	WSL2() bool
```

Comment:
I'm not sure i understand why we need this. If WSL2 is not compatible with cadvisor we should either: 
- Remove cadvisor (to remove a dependency on linux for e2e) 
- Keep cadvisor, and encourage people who use linux subsystem on linux to disable cadvisor. 

Adding a field to Environment configuration for this very specific edge case stinks of 'leaky abstraction'. 

Conversely you can invert this and have a 'useCadvisor' env variable that will determine if it should be enabled or not so there isn't this conceptual hop from: 
' Why is it relevant if i'm using WSL2?' 
*greps WSL2* 
'Oh, WSL doesn't support cadvisor.' 
'I'll just disable cadvisor'

Just my thoughts, not willing to die on this hill though. IMHO i'd rather avoid embedding OS checks in configuration for very specific feature compatibility.

Reply:
Got rid of the `WSL()` function in the `Environment` interface, as indeed I agree it leaks details from the `DockerEnvironment` type. Instead now I just import `e2einteractive` and use it from there. Note that as I mentioned in another comment, I would like to move the function to `efficientgo/core` if maintainers agree. Changes are in [4330eb9](https://github.com/efficientgo/e2e/pull/57/commits/4330eb9019e3b7a06be055b9873ddf59e7c66cea).

> Keep cadvisor, and encourage people who use linux subsystem on linux to disable cadvisor.

I talked with @bwplotka about this and our conclusion is that we should "panic" and stop execution in the scenario where we're running in WSL 2 and the option to run cAdvisor is set to `true`.